dimod/constrained/constrained.py

```diff
@@ -1144,7 +1144,11 @@ def from_file(cls,
 
         return cqm
 
-    def iter_constraint_data(self, sample_like: SamplesLike) -> Iterator[ConstraintData]:
+    def iter_constraint_data(self,
+                             sample_like: SamplesLike,
+                             *,
+                             labels: typing.Optional[typing.Iterable[typing.Hashable]] = None,
```

Comment:
The types seem to already be imported.
```suggestion
                             labels: Optional[Iterable[Hashable]] = None,
```

Reply:
Fair enough. I guess both is fine, although I generally find using `typing.` everywhere makes the types less legible, although there are benefits to it as well. Python's typing system could definitely be better.

Reply:
I agree it's more cluttered visually. But I prefer cluttered signatures to a cluttered namespace. Determining what's a local symbol vs an imported one is much nicer with the qualified names. With `typing` most folks probably know which is which, but I prefer to be explicit when possible.